src/core/manager.c

```diff
@@ -2284,6 +2284,32 @@ static int manager_dispatch_cgroups_agent_fd(sd_event_source *source, int fd, ui
         return 0;
 }
 
+static bool manager_process_barrier_fd(const char *buf, FDSet *fds) {
+        assert(buf);
+        assert(fds);
+
+        _cleanup_strv_free_ char **tags = NULL;
+
+        tags = strv_split(buf, NEWLINE);
+        if (!tags) {
+                log_oom();
+                return false;
+        }
+
+        if (strv_find(tags, "BARRIER=1")) {
+                /* when we process BARRIER=1, we process nothing else */
+                if (strv_length(tags) > 1)
+                        log_warning("Extra notification messages passed with BARRER=1, ignoring.");
```

Comment:
hmm, can you use log_unit_warning() here (i.e. which means accepting the `Unit` object as first arg). I am aware the code around here uses plain `log_warning()`, but let's get this right for new code. things are a ton more debuggable, if we see which service is misbehaving here...

also, type BARRER → BARRIER

Reply:
hmm, ignore the bit about log_unit_warning(). I missed that this code runs before we determine which unit this belongs to